var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -60,6 +61,19 @@ def install(self, spec, prefix):
         if spec.satisfies("@4.7.1:") and sys.platform != 'darwin':
             enabled_languages.add('go')
 
+        # Fix a standard header file for OS X Yosemite & earlier that
+        # is GCC incompatible by replacing non-GCC compliant macros
+        if sys.platform == 'darwin':
+            if isfile("/usr/include/dispatch/object.h"):
+                new_dispatch_dir = prefix + "include/dispatch"
+                mkdirp(new_dispatch_dir)
+                cp = which('cp')
+                cp("/usr/include/dispatch/object.h",
+                   new_dispatch_dir + "object.h")
+                filter_file("typedef void \(\^dispatch_block_t\)\(void\)",
```

Comment:
You might need an 'r' in front of the string so that the escape characters don't get interpreted.